code/src/hooks/use-infinite-scroll/useInfiniteScroll.test.ts

```diff
@@ -0,0 +1,65 @@
+import { renderHook } from "@testing-library/react";
+
+import useInfiniteScroll from "@/hooks/use-infinite-scroll/useInfiniteScroll";
+
+const mockAndRender = (item?: HTMLLIElement, isIntersecting: boolean = false) => {
+  const loadNextPage = jest.fn();
+  const { result } = renderHook(() => useInfiniteScroll(loadNextPage));
+
+  result.current(item ?? null);
+
+  if (item) {
+    const entry = { isIntersecting };
+    (global.IntersectionObserver as jest.Mock).mock.calls[0][0]([entry]);
```

Comment:
Try to create global `IntersectionObserver` mock because we use it also in `useStickyHeader` hook for example. We can add some code to `setup-teset.js` in this case

Reply:
Created, Thank you